fdbserver/RestoreApplier.actor.cpp

```diff
@@ -211,52 +213,76 @@ ACTOR static Future<Void> applyClearRangeMutations(Standalone<VectorRef<KeyRange
 	return Void();
 }
 
+ACTOR Future<Optional<Value>> getValue(Reference<ReadYourWritesTransaction> tr, Key key, int i,
+                                       std::set<int>* keysNotFound) {
+	try {
+		Optional<Value> v = wait(tr->get(key));
+		return v;
+	} catch (Error& e) {
+		if (e.code() == error_code_key_not_found) {
+			keysNotFound->insert(i);
+			return Optional<Value>();
+		} else {
+			throw;
+		}
+	}
+}
+
 // Get keys in incompleteStagingKeys and precompute the stagingKey which is stored in batchData->stagingKeys
 ACTOR static Future<Void> getAndComputeStagingKeys(
     std::map<Key, std::map<Key, StagingKey>::iterator> incompleteStagingKeys, double delayTime, Database cx,
     UID applierID, int batchIndex) {
 	state Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(cx));
-	state std::vector<Future<Optional<Value>>> fValues;
+	state std::vector<Future<Optional<Value>>> fValues(incompleteStagingKeys.size(), Never());
 	state int retries = 0;
+	state UID randomID = deterministicRandom()->randomUniqueID();
 
 	wait(delay(delayTime + deterministicRandom()->random01() * delayTime));
 	TraceEvent("FastRestoreApplierGetAndComputeStagingKeysStart", applierID)
+	    .detail("RandomUID", randomID)
 	    .detail("BatchIndex", batchIndex)
 	    .detail("GetKeys", incompleteStagingKeys.size())
 	    .detail("DelayTime", delayTime);
+	state std::set<int> keysNotFound;
+
+	state int i = 0;
 	loop {
 		try {
 			tr->reset();
```

Comment:
I think this `reset` is not right, because `onError` already resets the transaction state. If we `reset` here, then we lost the state for doing the exponential backoff.